src/api/v1/community_businesses/volunteer_logs.ts

```diff
@@ -99,6 +99,134 @@ const routes: Hapi.ServerRoute[] = [
     },
   },
 
+  {
+    method: 'POST',
+    path: '/community-businesses/me/volunteer-logs/sync',
+    options: {
+      description: 'Synchronise volunteer logs for own use at community business',
+      auth: {
+        strategy: 'standard',
+        access: {
+          // TO BE REPLACED: "parent" with "own"
+          scope: ['volunteer_logs-parent:write'],
+        },
+      },
+      validate: {
+        payload: Joi.array().items(Joi.object({
+          activity: Joi.string().required(),
+          duration: Joi.object({
+            hours: Joi.number().integer().min(0),
+            minutes: Joi.number().integer().min(0),
+            seconds: Joi.number().integer().min(0),
+          }).required(),
+          startedAt: Joi.date().iso().max('now').default(() => new Date().toISOString(), 'now'),
+        })).required(),
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request: SyncMyVolunteerLogsRequest, h) => {
+      const {
+        server: { app: { knex } },
+        auth: { credentials: { user } },
+        pre: { communityBusiness },
+        payload,
+      } = request;
+
+      try {
+        await knex.transaction(async (trx) =>
+          Promise.all(payload.map(async (log) =>
+            VolunteerLogs.add(knex, {
+              ...log,
+              organisationId: communityBusiness.id,
+              userId: user.id,
+            })
+          ))
+        );
+
+        return null;
+
+      } catch (error) {
+        if (error.code === '23502') { // Violation of null constraint implies invalid activity
+          return Boom.badRequest('Invalid activity');
+        }
+        if (error.code === '23505') { // Violation of unique constraint => duplicate log
+          return Boom.badRequest('Duplicate logs in payload, check start times');
+        }
+        throw error;
+      }
+    },
+  },
+
+  {
+    method: 'GET',
+    path: '/community-businesses/me/volunteer-logs/summary',
```

Comment:
Is there a difference between `summary` and `aggregates` endpoints? From what I've seen we have `cb../summary` and `users../aggregates` returning similar data

Reply:
I think I only used `summary` because that's what the old endpoint was called. I can switch to `aggregates`, no prob.

Reply:
I think I would prefer both of these endpoints to be named `/summary`. Then we can keep `/aggregates` for when we return dynamic aggregated data for the super admin dashboard.